client/app/src/js/services/services.js

```diff
@@ -66,6 +66,29 @@ function Screenshots($http, $location, $window) {
 };
 Screenshots.$inject = ['$http','$location','$window'];
 
+function UserInfo($http){
```

Comment:
Cool. That makes more sense. 

I'm not totally opposed to the idea, but want to raise one concern. There's something I find leery about having a stateful User factory. If the goal is to keep track of the currently authenticated User, does the JWT do that job? Currently, it doesn't look like the factory could be hacked -- the username is stored in a closure variable -- but it strikes me as redundant right now.